### PR TITLE
Adding controller to manage application cancellations

### PR DESCRIPTION
**Pull request by johnbrett**
This PR adds the angular controller and other necessary infrastructure to be able to handle the cancellation links from event application approval emails. Required for: https://github.com/CoderDojo/community-platform/issues/876

### CODE REVIEW

File: web/public/js/controllers/cancel-session-invite-controller.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    var application = {
+      id: applicationId,
+      eventId: eventId
+    };
+
+    cdEventsService.removeApplicant(application, function (response) {   
+        
+        var successMessage = $translate.instant('Thank you for confirming you will not be in attendance; your application has now been cancelled.');
```
Comment (tangentfairy) [inline web/public/js/controllers/cancel-session-invite-controller.js]: This string needs to be added to messages.po also

File: web/public/js/controllers/cancel-session-invite-controller.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          alertService.showAlert(successMessage);
+        } else {
+            alertService.showAlert($translate.instant(response.why));
+        }
+        $state.go('home');
+        
+    }, function (err) {
+        if (err) {
+            console.error(err);
+            alertService.showError($translate.instant('Invalid application cancellation link.'));
```
Comment (tangentfairy) [inline web/public/js/controllers/cancel-session-invite-controller.js]: Same for this one - needs to be added to messages.po!

**Comment by johnbrett** [comment]
@tangentfairy updated the PR :)

**Comment by tangentfairy** [comment]
Looks good thanks @johnbrett ! :+1:

**Pull request merged**